Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Collections/ObjectCollectionNode.cs

```diff
@@ -10,11 +10,15 @@ namespace Microsoft.MixedReality.Toolkit.Utilities
     /// Collection node is a data storage class for individual data about an object in a collection.
     /// </summary>
     [Serializable]
-    public struct ObjectCollectionNode
+    public class ObjectCollectionNode
     {
         public string Name;
         public Vector2 Offset;
         public float Radius;
         public Transform Transform;
+        public GameObject GameObject;
+        public Collider Collider;
+        public Collider[] Colliders;
+        public bool isClipped;
```

Comment:
@Cameron-Micka FYI on the `MaterialInstance` issues. Also, that error is coming back.

Reply:
I'm guessing that error is from a material that was already instanced being instanced a second time. We should figure out what material(s) those are? I should demote the error to a warning and print out the material(s) in question. As for the frame time that looks like there are ~278 renderers in the clipping primitive. So, it has to loop though them all and update their material property blocks. Same is true for the MaterialInstance class. There are ~278 instances all their Update being called. I would recommended simplifying the scene a bit. In the future we could add some logic to the ClippingPrimitive to not update if no changes were made to the primitive. But, I don't think that should block this PR.

Reply:
Cool, I second Camerons opinion. It is worrisome and important but should not block this PR.

@Cameron-micka, can you open up a bug with all relevant details for the MaterialInstance update perf issue since you seem to have a solid grasp of it?

If you also find noteworthy details for clipping primitive as well, please file a bug

Reply:
Of course! Added some bugs: https://github.com/microsoft/MixedRealityToolkit-Unity/issues/6270 and https://github.com/microsoft/MixedRealityToolkit-Unity/issues/6271